admin-frontend/open_admin_app/lib/routes/create_admin_api_keys_route.dart

```diff
@@ -0,0 +1,246 @@
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: const <Widget>[TopWidget(), BottomWidget()],
+    );
+  }
+}
+
+class TopWidget extends StatelessWidget {
+  const TopWidget({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final bloc = BlocProvider.of<CreateUserBloc>(context);
+
+    return StreamBuilder<CreateUserForm>(
+        stream: bloc.formState,
+        builder: (context, AsyncSnapshot<CreateUserForm> snapshot) {
+          if (snapshot.hasData &&
+              snapshot.data == CreateUserForm.successState) {
+            return const TopWidgetSuccess();
+          }
+          // ignore: prefer_const_constructors
+          return TopWidgetDefault();
+        });
+  }
+}
+
+class TopWidgetDefault extends StatefulWidget {
+  const TopWidgetDefault({Key? key}) : super(key: key);
+
+  @override
+  _TopWidgetDefaultState createState() => _TopWidgetDefaultState();
+}
+
+class _TopWidgetDefaultState extends State<TopWidgetDefault> {
+  final _name = TextEditingController();
+  bool isAddButtonDisabled = true;
+
+  @override
+  void didUpdateWidget(TopWidgetDefault oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    final bloc = BlocProvider.of<CreateUserBloc>(context);
+    bloc.formKey = GlobalKey<FormState>();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final bloc = BlocProvider.of<CreateUserBloc>(context);
+
+    return Form(
+        key: bloc.formKey,
+        child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              const FHHeader(title: 'Create new Admin API Key'),
+              Container(
+                constraints: const BoxConstraints(maxWidth: 300),
+                child: Column(
+                  children: [
+                    const SizedBox(height: 16.0),
+                    TextFormField(
+                      controller: _name,
+                      decoration: fhFilledInputDecoration(
+                        labelText: 'Name',
+                      ),
+                      validator: (v) {
+                        if (v?.isEmpty == true) {
+                          return 'Please provide a name for the Admin API Key';
+                        }
+                        return null;
+                      },
+                      onSaved: (v) => bloc.name = v,
+                    ),
+                  ],
+                ),
+              ),
+              Padding(
+                padding: const EdgeInsets.only(top: 30.0),
+                child: Text(
+                  'Assign to some portfolio groups or leave it blank to add them later',
+                  style: Theme.of(context).textTheme.caption,
+                ),
+              ),
+              const PortfolioGroupSelector(),
+            ]));
+  }
+}
+
+class TopWidgetSuccess extends StatelessWidget {
```

Comment:
these names are now duplicated, they need to be unique or prefixed with an _ to keep them private to this file.